File: src/enums.py
"""Enumerations used across the application.

This module contains all enum types shared between models, schemas, and services.
Keeping enums separate avoids circular dependencies and maintains clean architecture.
"""

from enum import Enum


class UserRole(str, Enum):
    """User role enumeration."""

    ADMIN = "ADMIN"
    COMPANY = "COMPANY"


class JobStatus(str, Enum):
    """Job status enumeration."""

    PENDING_APPROVAL = "PENDING_APPROVAL"
    PUBLISHED = "PUBLISHED"
    CLOSED = "CLOSED"


class ApplicationStatus(str, Enum):
    """Application (Match) status enumeration."""

    NEW = "NEW"
    APPROVED_BY_ADMIN = "APPROVED_BY_ADMIN"
    REJECTED = "REJECTED"
    HIRED = "HIRED"

File: frontend/src/types/enums.ts
/**
 * Enum-equivalent const objects mirroring backend src/enums.py.
 * Uses `as const` objects instead of TS enums — required by TS 6.0
 * with erasableSyntaxOnly enabled (enums emit runtime code).
 */

export const UserRole = {
  ADMIN: "ADMIN",
  COMPANY: "COMPANY",
  CANDIDATE: "CANDIDATE",
} as const;
export type UserRole = (typeof UserRole)[keyof typeof UserRole];

export const JobStatus = {
  PENDING_APPROVAL: "PENDING_APPROVAL",
  PUBLISHED: "PUBLISHED",
  CLOSED: "CLOSED",
} as const;
export type JobStatus = (typeof JobStatus)[keyof typeof JobStatus];

export const ApplicationStatus = {
  NEW: "NEW",
  APPROVED_BY_ADMIN: "APPROVED_BY_ADMIN",
  REJECTED: "REJECTED",
  HIRED: "HIRED",
  WITHDRAWN: "WITHDRAWN",
} as const;
export type ApplicationStatus =
  (typeof ApplicationStatus)[keyof typeof ApplicationStatus];

export const InviteTokenStatus = {
  PENDING: "PENDING",
  USED: "USED",
  EXPIRED: "EXPIRED",
  REVOKED: "REVOKED",
} as const;
export type InviteTokenStatus =
  (typeof InviteTokenStatus)[keyof typeof InviteTokenStatus];
